Project Management Process.md

```diff
@@ -0,0 +1,287 @@
+    Milestone A     |     Milestone 2           Goal C
+    Milestone B     |       Goal 4              Goal D
+    Milestone C     |     Milestone 3      
+                    |       Goal 5      
+  Projects          |  
+    Project 1       |   Backlog             Backlog
+    Project 2       |     Goal 1              Goal A
+    Project 3       |     Goal 2              Goal B
+                    |     Goal 3              Goal C
+                    |     Goal 4              Goal D
+                    |     Goal 5
+```
+
+### Goals
+At the heart of each project are the goals. They are the basic unit of work. Goals can be anything that ***adds value to a release***. This includes issues reported by users, feature requests, bugs, Pull-Requests, general tasks such as refactoring, documentation or research. Goals are collected, maintained and tracked in the backlog by the project lead.
+
+A goal description includes the following:
+- Name / Summary
+  - Describes what the goal is on high level
+  - Example 1: "Reduce the size of the go-ipfs executable"
+  - Example 2: "js-ipfs should have API Documentation"
+- Description
+  - What the problem is and how to achieve this goal
+  - Should tell you how you will know that the problem has been solved or the goal has been achieved
+  - Describe the type of this goal: is it a bug? a new feature? something else?
+- Status
+  - The status should signal the goal's status towards completion
+  - What is the current state of this goal? Is it to-be-reviewed? In progress? Blocked? Done?
+  - A goal can be in only one state at a time
+- Dependencies (where applicable)
+  - Links to other goals that must happen before or concurrently with this one
+- Any additional information that helps to reach this goal (optional)
+
+TODO: a clear example
+
+### Backlog
+Backlog is the collection of all goals in a project. It should be as comprehensive as possible, including everything that needs to happen or should happen in a particular project. The backlog indicates priority by order: the items that have the highest priority are at the top of the list. The backlog is owned and maintained by the project lead. As part of daily work, the backlog gets updated and a general grooming should happen on weekly basis.
+
+A backlog has the following information:
+- All goals of a project
+- Priority of goals in sorted order
+
+While the backlog is usually one big list of goals, it may sometimes become convoluted. In this case, the backlog list can be split in two: ***Backlog*** and ***Future Work***. The future work section lists goals that will probably not be worked on anytime soon, or are not ready to be worked on yet, whereas the ready section contains the goals that are ready to be worked on or "approved". If split, together they are called "the backlog" while *Future Work* is referenced only in its specific meaning. The split and maintenance of both lists are done by the project lead.
```

Comment:
Not quite, let me clarify my point of view then.

It is important to categorize our project visitors with regards to what they are looking for when they come to 'visit' the project. I believe there are mainly two classes (that some groups may overlap over the two):

**The Users/Consumers** 

The groups that want to use project X or are already using project X and therefore want to know when they can have the feature Y, so for that, the Roadmap with milestones is perfect 👌.

**The Hackers/Contributors**

The community members that want to help the project form itself, that want to learn with it and have energy available to invest in the development of the project. These groups will appreciate the Roadmap, however, when they will need to understand how these Roadmap milestone/goal items map to each repository (i.e. the collection of issues), so they can understand more how to participate on that specific endeavour.

To solve this, what I did for js-ipfs was creating issues to track each of this goals (e.g https://github.com/ipfs/js-ipfs/issues/58, https://github.com/ipfs/js-ipfs/issues/51, https://github.com/ipfs/js-ipfs/issues/60, https://github.com/ipfs/js-ipfs/issues/92) so that I had a live thread for each goal inside milestones. This seemed to work well. It also enabled me to tag issues with difficulty and what was 'blocked' or 'unblocked', it was just great 👌

Now, coming from another way, instead of being a person that just eagerly wants to hack on something, actually has an issue that they would like to get solved (or an example they would like to see written), but unfortunately, that is not part of our Roadmap until the 4 milestones after. It might be clear for us that issue needs to be postponed, but since we only have a way to label it 'backlog' or 'future work', it might not be clear if we 'froze it' or how long are we deferring it.

These last two points is where I feel something is missing from the pm doc. In js-ipfs, I felt the need to patch that hole by creating the tracking issues and by labelling issues with the milestone numbers.

------ 

Here is an idea, we define a metric, like 'number of contributions' over a unit of time of length Z and we see how that metric evolves. My guess is that each project Captain will have specific last touches when they apply the Model to their Mechanics, once those are in place and we have metrics, then we can revisit what makes a project really good to invite collaborators in and understand how we can do to optimize for that across the board.